lib/OpenQA/Git/MaintenanceOutage.pm

```diff
@@ -0,0 +1,31 @@
+package OpenQA::Git::MaintenanceOutage;
+use Mojo::Base -strict, -signatures;
+use Mojo::File 'path';
+use Time::Seconds;
+
+my $DEFAULT_STATE_FILE = "/var/lib/openqa/share/gitlab_maintenance.lock";
+my $MAX_DURATION_SECONDS = 1800;
```

Comment:
This should probably be a constant and be overridable by an env variable (like we do in many places).